autoload/ale/lsp/response.vim

```diff
@@ -27,8 +27,13 @@ function! ale#lsp#response#ReadDiagnostics(response) abort
 
     for l:diagnostic in a:response.params.diagnostics
         let l:severity = get(l:diagnostic, 'severity', 0)
+        " we want to preserve the line breaks for `detail` but replace with
+        " spaces for text
+        let l:diagnostic.detail = l:diagnostic.message
+        let l:diagnostic.text = substitute(l:diagnostic.message, "\n", ' ', 'g')
```

Comment:
Don't modify the incoming object here, just adjust the objects we're creating.

Reply:
Do you mean something like:
```
        let l:diagnostic.text = l:diagnostic.message
        let l:diagnostic.text = substitute(l:diagnostic.text, "\n", ' ', 'g')
```

I don't think this passes the unit test. I am not very good at Vimscript so sorry about these problems.

Reply:
You can pass `substitute(l:diagnostic.message, "\n", ' ', 'g')` to the text key below without modifying the object, and use `l:diagnostic.message` below that as before.